imagetest/test_suites/image_validation/cloud_test.go

```diff
@@ -0,0 +1,97 @@
+package imagevalidation
+
+import (
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+const (
+	chronyService = "chronyd"
+	ntpdService   = "ntp"
+)
+
+var ntpConfig = []string{"/etc/ntp.conf"}
+var chronyConfig = []string{"/etc/chrony.conf", "/etc/chrony/chrony.conf", "/etc/chrony.d/gce.conf"}
+
+// TestNTPService Verify that ntp package exist and configuration is correct.
+// For SUSE, the ntp service is ntpd. For others, the ntp service is chronyd
+func TestNTPService(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("Couldn't get image from metadata")
+	}
+	var servicename string
+	var configPaths []string
+	switch {
+	case strings.Contains(image, "debian-10"):
+	case strings.Contains(image, "debian-11"):
+	case strings.Contains(image, "rhel"):
+	case strings.Contains(image, "centos"):
+	case strings.Contains(image, "sles"):
+		servicename = chronyService
+		configPaths = chronyConfig
+	case strings.Contains(image, "ubuntu"):
+		switch {
+		case strings.Contains(image, "ubuntu-1604"):
+		case strings.Contains(image, "ubuntu-minimal-1604"):
+			servicename = ntpdService
+			configPaths = ntpConfig
+		default:
+			servicename = chronyService
+			configPaths = chronyConfig
+		}
```

Comment:
i think this would be more readable if you didn't nest them. raise out the explicit ubuntu tests so they match first, then the general ubuntu match afterward where you have 'default' here

Reply:
I move the nest out and make the default for chrony which is more general